src/jsonpatcherproxy.js

```diff
@@ -139,37 +142,42 @@ const JSONPatcherProxy = (function() {
     if (Array.isArray(target) && !Number.isInteger(+key.toString())) {
       return Reflect.set(target, key, newValue);
     }
+    let reflectionResult;
     if (target.hasOwnProperty(key)) {
       if (typeof target[key] == 'undefined') {
         if (Array.isArray(target)) {
+          reflectionResult = Reflect.set(target, key, newValue);
           instance.defaultCallback({
             op: 'replace',
             path: destinationPropKey,
             value: newValue
           });
         } else {
+          reflectionResult = Reflect.set(target, key, newValue);
           instance.defaultCallback({
             op: 'add',
             path: destinationPropKey,
             value: newValue
           });
         }
-        return Reflect.set(target, key, newValue);
+        return reflectionResult;
       } else {
+        reflectionResult = Reflect.set(target, key, newValue);
         instance.defaultCallback({
           op: 'replace',
           path: destinationPropKey,
           value: newValue
         });
-        return Reflect.set(target, key, newValue);
+        return reflectionResult;
       }
     } else {
+      reflectionResult = Reflect.set(target, key, newValue);
       instance.defaultCallback({
         op: 'add',
         path: destinationPropKey,
         value: newValue
       });
-      return Reflect.set(target, key, newValue);
+      return reflectionResult;
```

Comment:
`setTrap` has several calls to `Reflect.set`, `instance.defaultCallback`. The only thing that changes is the config object that you pass to `instance.defaultCallback`.

A more DRY thing to do would be to keep a single call at the end of `setTrap`:

```
const reflectionResult = Reflect.set(target, key, newValue);
instance.defaultCallback(patch);return reflectionResult;
```